app/Library/UserService.php

```diff
@@ -35,11 +35,17 @@ public function attachInstructorsToCourses(Collection $courses): Collection {
         });
         
     
-        return $courses->each(function ($course) {
+        return $courses->each(function ($course) use ($employeeList){
             if (!$course->INSTRUCTOR_EMPLID) return;
 
             $user = $this->findOrCreateByEmplid($course->INSTRUCTOR_EMPLID);
+            $employeeInfo = $employeeList->where('EMPLID', $user->emplid)->first();
```

Comment:
- `$user` can possible be null (e.g. they aren't in ldap), so this line will sometimes throw.
- might be worth creating a keyed employee list before the loop?

Reply:
k, fixed the null user. Is your thinking on the keyed array just for performance differences vs however eloquent implements where() on a collection?

Reply:
Yeah, just for perf. Might not matter much though or maybe better solutions.